WalletWasabi/Wallets/P2PNodesManager.cs

```diff
@@ -52,6 +52,7 @@ public async Task<Node> GetNodeForSingleUseAsync(CancellationToken cancellationT
 
 	public void DisconnectNodeIfEnoughPeers(Node node, string reason)
 	{
+		_nodesInUse.Remove(node);
```

Comment:
This will remove the node even if there are not enough nodes. Isn't that wrong?

Reply:
No. That's was the bug.